plugins/trendmicro_visionone/icon_trendmicro_visionone/triggers/poll_alert_list/trigger.py

```diff
@@ -0,0 +1,51 @@
+import insightconnect_plugin_runtime
+import time
+from .schema import PollAlertListInput, PollAlertListOutput, Input, Output, Component
+from insightconnect_plugin_runtime.exceptions import PluginException
+
+# Custom imports below
+import json
+
+
+class PollAlertList(insightconnect_plugin_runtime.Trigger):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="poll_alert_list",
+            description=Component.DESCRIPTION,
+            input=PollAlertListInput(),
+            output=PollAlertListOutput(),
+        )
+
+    def run(self, params={}):
+        """Run the trigger"""
+        # Get Connection Client
+        client = self.connection.client
+        # Get Action Parameters
+        start_date_time = params.get(Input.START_DATE_TIME)
+        end_date_time = params.get(Input.END_DATE_TIME)
+
+        while True:
+            new_alerts = []
+            # Make Action API Call
+            self.logger.info("Making API Call...")
+            try:
+                client.consume_alert_list(
+                    lambda alert: new_alerts.append(alert.json()),
+                    start_time=start_date_time,
+                    end_time=end_date_time,
+                )
+            except Exception as error:
+                raise PluginException(
+                    cause="An error occurred while polling alerts.",
+                    assistance="Please check your inputs and try again.",
+                    data=error,
+                )
+            # Load json objects to list
+            alert_list = []
+            for i in new_alerts:
+                alert_list.append(json.loads(i))
+            # Return results
+            self.logger.info("Returning Results...")
+            self.send({Output.TOTAL_COUNT: len(new_alerts), Output.ALERTS: alert_list})
```

Comment:
time.sleep(1800) will sleep for 30 minutes, not 1,8 seconds.
And yes this is intentional, because the primary purpose of these triggers, as we theorized, would be to input a future end_date_time, and so have the trigger re-run itself and return any new alerts it would find, as well as the same ones, since they are there in the same time-frame. But also if they are to be used in a workflow in a past time-frame then they should return exactly what alerts are in that past time-frame.